### DISCUSSION TITLE
Need to wait longer for proxmox_virtual_environment_vm's ipv4_addresses

### DISCUSSION THREAD

**Discussion by dandaolrian** [bpg/terraform-provider-proxmox#1450] [General]
We've run into something similar to what was discussed in #100. 
In our case we've just run into a timing problem whereby our remote-exec's were failing as the call to get network interfaces seems too early for our DHCP IP to be picked up.

Instead we got the "valid" unicast IP from Hyper-V adapter, but this isn't usable with our subsequent winrm provisioner step.
We checked the code and found there is a hard-coded 5 second wait before trying to get all network interfaces. If we bumped this up to 30 seconds we were OK. Could be this also would help with #776

Ideas for solutions:
1. So quick fix we could create a pull request to make that delay from function WaitForNetworkInterfacesFromVMAgent configurable, like the timeouts of the vm resource.
2. An alternative idea would be to check for each network device defined mac address and wait until all mac adresses have a matching IP. That assumes somewhere you get the mac address back somewhere else when not set

Any thoughts / better suggestions

**Comment by bpg-dev** [comment]
More refs: https://github.com/bpg/terraform-provider-proxmox/issues/668#issuecomment-1785831601
